crates/uv-tool/src/lib.rs

```diff
@@ -367,20 +376,16 @@ impl fmt::Display for InstalledTool {
 /// Errors if a directory cannot be found.
 pub fn find_executable_directory() -> Result<PathBuf, Error> {
     std::env::var_os("UV_TOOL_BIN_DIR")
-        .and_then(dirs_sys::is_absolute_path)
-        .or_else(|| std::env::var_os("XDG_BIN_HOME").and_then(dirs_sys::is_absolute_path))
+        .and_then(is_absolute_path)
+        .or_else(|| std::env::var_os("XDG_BIN_HOME").and_then(is_absolute_path))
         .or_else(|| {
             std::env::var_os("XDG_DATA_HOME")
-                .and_then(dirs_sys::is_absolute_path)
+                .and_then(is_absolute_path)
                 .map(|path| path.join("../bin"))
         })
         .or_else(|| {
-            // See https://github.com/dirs-dev/dirs-rs/blob/50b50f31f3363b7656e5e63b3fa1060217cbc844/src/win.rs#L5C58-L5C78
-            #[cfg(windows)]
-            let home_dir = dirs_sys::known_folder_profile();
-            #[cfg(not(windows))]
-            let home_dir = dirs_sys::home_dir();
-            home_dir.map(|path| path.join(".local").join("bin"))
+            let home_dir = etcetera::home_dir();
+            home_dir.map(|path| path.join(".local").join("bin")).ok()
```

Comment:
This seems to be _mostly_ the same except that it defers to `USERPROFILE` first?

Reply:
@zanieb -- I think my preference here is just to ship this as part of 0.5.0 in the near-ish future, and accept that we now respect non-empty `USERPROFILE` on Windows (which seems fine).